### PR TITLE
Some doc improvements to broadcast

### PR DESCRIPTION
**Pull request by kshyatt**
Everyone loves a little more cross-referencing.

### CODE REVIEW

File: base/broadcast.jl [hunk 1]
```diff
@@ -267,9 +267,9 @@ instantiate(x) = x
 
 Construct and check the axes for the lazy Broadcasted object `bc`.
 
-Custom `BroadcastStyle`s may override this default in cases where it is fast and easy
+Custom [`BroadcastStyle`](@ref)s may override this default in cases where it is fast and easy
```
Comment (ararslan) [inline base/broadcast.jl]: I don't think there are docs for `BroadcastStyle`.

Comment (kshyatt) [reply base/broadcast.jl]: https://github.com/JuliaLang/julia/blob/master/base/broadcast.jl#L112 ?
Appears in the docs: https://docs.julialang.org/en/latest/base/arrays/#Base.Broadcast.BroadcastStyle

Comment (ararslan) [reply base/broadcast.jl]: Oh, okay. Never mind then.

**Review by ararslan — APPROVED** [review]

**Comment by ViralBShah** [comment]
Rebased on master to get the doctest fix.

**Comment by fredrikekre** [comment]
Docs build fine locally.

**Pull request merged**